tools/gcp/utils/cleanup_xds_resources.py

```diff
@@ -0,0 +1,174 @@
+                f'test-backend-service-alternate{suffix}', '--global')
+    exec_gcloud('compute', 'backend-services', 'delete',
+                f'test-backend-service-extra{suffix}', '--global')
+    exec_gcloud('compute', 'backend-services', 'delete',
+                f'test-backend-service-more-extra{suffix}', '--global')
+    exec_gcloud('compute', 'firewall-rules', 'delete', f'test-fw-rule{suffix}')
+    exec_gcloud('compute', 'health-checks', 'delete', f'test-hc{suffix}')
+    exec_gcloud('compute', 'instance-groups', 'managed', 'delete',
+                f'test-ig{suffix}', '--zone', ZONE)
+    exec_gcloud('compute', 'instance-groups', 'managed', 'delete',
+                f'test-ig-same-zone{suffix}', '--zone', ZONE)
+    exec_gcloud('compute', 'instance-groups', 'managed', 'delete',
+                f'test-ig-secondary-zone{suffix}', '--zone', SECONDARY_ZONE)
+    exec_gcloud('compute', 'instance-templates', 'delete',
+                f'test-template{suffix}')
+
+
+def remove_relative_resources_psm_sec(prefix: str):
+    """Removing GCP resources created by PSM Sec framework."""
+    logging.info('Removing PSM Security resources with prefix [%s]', prefix)
+    exec_gcloud('compute', 'forwarding-rules', 'delete',
+                f'{prefix}-forwarding-rule', '--global')
+    exec_gcloud('alpha', 'compute', 'target-grpc-proxies', 'delete',
+                f'{prefix}-target-proxy')
+    exec_gcloud('compute', 'url-maps', 'delete', f'{prefix}-url-map')
+    exec_gcloud('compute', 'backend-services', 'delete',
+                f'{prefix}-backend-service', '--global')
+    exec_gcloud('compute', 'health-checks', 'delete', f'{prefix}-health-check')
+    exec_gcloud('compute', 'firewall-rules', 'delete',
+                f'{prefix}-allow-health-checks')
+    exec_gcloud('alpha', 'network-security', 'server-tls-policies', 'delete',
+                f'{prefix}-server-tls-policy')
+    exec_gcloud('alpha', 'network-security', 'client-tls-policies', 'delete',
+                f'{prefix}-client-tls-policy')
+
+
+def check_one_type_of_gcp_resources(list_cmd: List[str],
+                                    suffix_search: str = '',
+                                    prefix_search: str = ''):
+    logging.info('Checking GCP resources with %s or %s', suffix_search,
+                 prefix_search)
+    for resource in exec_gcloud(*list_cmd):
+        if resource['name'].startswith('interop-psm-url-map'):
+            logging.info('Skipping url-map test resource: %s', resource['name'])
+            continue
+        if suffix_search:
+            result = re.search(suffix_search, resource['name'])
+            if result is not None:
+                remove_relative_resources_run_xds_tests(result.group(1))
+                continue
+
+        if prefix_search:
+            result = re.search(prefix_search, resource['name'])
+            if result is not None:
+                remove_relative_resources_psm_sec(result.group(1))
```

Comment:
Should we potentially wait for @menghanl's new script? It should support more interop tests than security, and also take care of k8s. Or we could hide this one behind a flag.

Reply:
Let's go with this PR. I will update it to handle resources for the new script